library/L1_Peripheral/lpc40xx/test/capture_test.cpp

```diff
@@ -0,0 +1,153 @@
+       Method(mock_pin1, SetPinFunction));
+
+    sjsu::Pin & capture0_input_pin = mock_pin0.get();
+    sjsu::Pin & capture1_input_pin = mock_pin1.get();
+
+    static const Capture::CaptureChannelPartial_t kTestTimerPartial =
+    {
+        .timer_register = &test_timer_register,
+        .power_id = SystemController::Peripherals::kTimer0,
+        .irq = IRQn::TIMER0_IRQn,
+        .user_callback = &test_timer_isr,
+        .capture_pin0 = capture0_input_pin,
+        .capture_pin1 = capture1_input_pin
+    };
+
+    const Capture::CaptureChannel_t kTestTimer =
+    {
+        .channel = kTestTimerPartial,
+        .isr = sjsu::lpc40xx::Capture::TimerHandler<kTestTimerPartial>
+    };
+
+    memset(&test_timer_register, 0, sizeof(test_timer_register));
+
+    Mock<sjsu::SystemController> mock_system_controller;
+    Fake(Method(mock_system_controller, PowerUpPeripheral));
+
+    constexpr units::frequency::hertz_t kTestSystemFrequency = 4_MHz;
+    constexpr int kTestPeripheralClockDivider = 1;
+
+    When(Method(mock_system_controller, GetSystemFrequency))
+        .AlwaysReturn(kTestSystemFrequency);
+    When(Method(mock_system_controller, GetPeripheralClockDivider))
+        .AlwaysReturn(kTestPeripheralClockDivider);
+
+    Mock<sjsu::InterruptController> mock_interrupt_controller;
+    Fake(Method(mock_interrupt_controller, Register));
+
+    Capture test_subject(kTestTimer,
+        mock_system_controller.get(),
+        mock_interrupt_controller.get());
+
+    SECTION("Capture Initialize")
+    {
+        constexpr units::frequency::hertz_t kTestFrequency = 1_MHz;
+
+        test_subject.Initialize(kTestFrequency, &DummyCallback);
+        int prescaler = (kTestSystemFrequency.to<int32_t>() /
+                kTestPeripheralClockDivider) / kTestFrequency.to<int32_t>();
+
+        Verify(Method(mock_interrupt_controller, Register)
+        .Matching(
+            [kTestTimer](
+                sjsu::InterruptController::RegistrationInfo_t info) {
+              return (info.interrupt_request_number ==
+                      kTestTimer.channel.irq) &&
+                    (info.interrupt_service_routine ==
+                      kTestTimer.isr) &&
+                    (info.enable_interrupt == true) &&
+                    (info.priority == -1);
+            }));
+
+        CHECK(test_timer_register.PR == prescaler);
+        CHECK(bit::Read(test_timer_register.TCR, 0) == true);
+    }  // end section initialize
+
+    SECTION("Capture Config")
+    {
+      for (size_t channel = 0; channel < std::size(kChannelList); channel++)
+      {
+        int position = 0;
+        if (channel)
+        {
+          position = 3;
+        }
+
+        for (size_t mode = 0; mode < std::size(kModeList); mode++)
+        {
+          test_subject.ConfigureCapture(kChannelList[channel], kModeList[mode]);
+          CHECK(bit::Extract(test_timer_register.CCR, position, 2) ==
+                kModeList[mode]);
+        }
+        }
```

Comment:
Incorrect indentation for the curly brace. Please use `make format-code` to fix this.